pkg/workloadslicing/workloadslicing.go

```diff
@@ -125,8 +126,18 @@ func FindNotFinishedWorkloads(ctx context.Context, clnt client.Client, jobObject
 	})
 
 	// Filter out workloads with activated "Finished" condition.
-	return slices.DeleteFunc(list.Items, func(w kueue.Workload) bool {
+	result := slices.DeleteFunc(list.Items, func(w kueue.Workload) bool {
 		return apimeta.IsStatusConditionTrue(w.Status.Conditions, kueue.WorkloadFinished)
+	})
+	replacedSlices := sets.New[workload.Reference]()
+	for _, w := range result {
+		if replacedKey := ReplacementForKey(&w); replacedKey != nil {
+			replacedSlices.Insert(*replacedKey)
+		}
+	}
+	// Filter out workloads with activated "Finished" condition.
+	return slices.DeleteFunc(result, func(w kueue.Workload) bool {
+		return replacedSlices.Has(workload.Key(&w))
 	}), nil
```

Comment:
```suggestion
	return slices.DeleteFunc(list.Items, func(w kueue.Workload) bool {
		return ReplacementForKey(&w) == nil || apimeta.IsStatusConditionTrue(w.Status.Conditions, kueue.WorkloadFinished)
	}), nil
```

why do we need additional loops?

Reply:
Ah, I see. The list.Items contain both old and new WorkloadSlices.

In that case, couldn't we do the following?

```suggestion
	replacedSlices := sets.New[workload.Reference]()
	for _, wl := range list.Items {
		if replacedKey := ReplacementForKey(&wl); replacedKey != nil {
			replacedSlices.Insert(*replacedKey)
		}
	}
	return slices.DeleteFunc(list.Items, func(w kueue.Workload) bool {
		return replacedSlices.Has(workload.Key(&w)) || apimeta.IsStatusConditionTrue(w.Status.Conditions, kueue.WorkloadFinished)
	}), nil
```

Reply:
https://github.com/kubernetes-sigs/kueue/pull/6203/files#r2236142599

Reply:
This will work too